NomicFoundation/hardhat-vscode
# Feat/bundle extension

A new package.json scripts command `package`, will clean, bundle the extension code and create the `.vsix` file.

Esbuild is used and added as a dev dependency for the bundling and minification.

To meet the references in package.json, we replace the files in the client and server out directories in place (so they are in the same place they would be during integration testing and debugging).

## Preview

![image](https://user-images.githubusercontent.com/24030/155779748-856ffd2a-7689-4391-9e88-ab441e1f9c78.png)

## Manual testing

Build a new `.vsix` file with:

```shell
yarn package
```

Install the generated `vsix` file into your vscode instance and check:

1. The extension installs
2. On opening a `.sol` file, you get highlighting
3. On making an edit you can get a warning/error from the compiler (this stresses, the `helper.js` part of the bundling).

## Considerations

The layout internal to the `vsix` matches up with the structure of our repo and its out folders, it proved too painful to have one structure for the extension and another for the repo.

We could cut the extension size in half if we could remove the bundling of `prettier`.
